core/port/range.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package port
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Range represents a networking port range
+type Range struct {
+	Start, End int
+}
+
+// UnspecifiedRange creates an unspecified range
+func UnspecifiedRange() *Range {
+	return &Range{0, 0}
+}
+
+// ParseRange parses port range expression, e.g. "1000:1200" into Range struct
+func ParseRange(rangeExpr string) (*Range, error) {
+	if rangeExpr == "" {
+		return UnspecifiedRange(), nil
+	}
+	bounds := strings.FieldsFunc(rangeExpr, func(c rune) bool {
```

Comment:
Maybe [strings.Split()](https://golang.org/pkg/strings/#Split) will be a better choice?

Reply:
If using `split` you have to deal with empty tokens

Reply:
Since I'm converting tokens with `Atoi` and checking for err anyway, I think it's fine to let the empty tokens slip here. Changing to `Split()`